modules/fedimint-ln-client/src/lib.rs

```diff
@@ -664,7 +688,7 @@ impl LightningClientModule {
         let contract_amount_msat = invoice_amount_msat + base_fee + margin_fee;
         let contract_amount = Amount::from_msats(contract_amount_msat);
 
-        let user_sk = bitcoin::KeyPair::new(&self.secp, &mut rng);
+        let user_sk = self.outgoing_redeem_key;
```

Comment:
This is a huge privacy regression. A new key should be derived each time.

Reply:
This should be fixed now. Similar to incoming contracts, each time an outgoing contract is created, we increment an index that is saved to the database so that we use a different key per outgoing contract.